algorithms/src/sorting/Kokkos_SortPublicAPI.hpp

```diff
@@ -77,130 +30,30 @@ namespace Kokkos {
-    auto const max_val = static_cast<double>(result.max_val);
-    auto const min_val = static_cast<double>(result.min_val);
-    // using 10M as the cutoff for special behavior (roughly 40MB for the count
-    // array)
-    if ((max_val - min_val) < 10000000) {
-      max_bins     = max_val - min_val + 1;
-      sort_in_bins = false;
-    }
-  }
-  if (std::is_floating_point<typename ViewType::non_const_value_type>::value) {
-    KOKKOS_ASSERT(std::isfinite(static_cast<double>(result.max_val) -
-                                static_cast<double>(result.min_val)));
-  }
-
-  using CompType = BinOp1D<ViewType>;
-  BinSort<ViewType, CompType> bin_sort(
-      view, CompType(max_bins, result.min_val, result.max_val), sort_in_bins);
-  bin_sort.create_permute_vector(exec);
-  bin_sort.sort(exec, view);
-}
-
-#if defined(KOKKOS_ENABLE_ONEDPL)
-template <class DataType, class... Properties>
-void sort(const Experimental::SYCL& space,
+void sort([[maybe_unused]] const ExecutionSpace& exec,
           const Kokkos::View<DataType, Properties...>& view) {
+  // constraints
   using ViewType = Kokkos::View<DataType, Properties...>;
-  static_assert(SpaceAccessibility<Experimental::SYCL,
-                                   typename ViewType::memory_space>::accessible,
-                "SYCL execution space is not able to access the memory space "
-                "of the View argument!");
-
-  // Can't use Experimental::begin/end here since the oneDPL then assumes that
-  // the data is on the host.
-  static_assert(
-      ViewType::rank == 1 &&
-          (std::is_same<typename ViewType::array_layout, LayoutRight>::value ||
-           std::is_same<typename ViewType::array_layout, LayoutLeft>::value),
-      "SYCL sort only supports contiguous rank-1 Views.");
-
-  if (view.extent(0) == 0) {
-    return;
-  }
-
-  auto queue  = space.sycl_queue();
-  auto policy = oneapi::dpl::execution::make_device_policy(queue);
-  const int n = view.extent(0);
-  oneapi::dpl::sort(policy, view.data(), view.data() + n);
-}
-#endif
-
-// clang-format off
-template <class ExecutionSpace, class DataType, class... Properties>
-std::enable_if_t<
-  (Kokkos::is_execution_space<ExecutionSpace>::value) &&
-  (SpaceAccessibility<
-     HostSpace, typename Kokkos::View<DataType, Properties...>::memory_space
-    >::accessible)
-  >
-// clang-format on
-sort(const ExecutionSpace&, const Kokkos::View<DataType, Properties...>& view) {
-  using ViewType = Kokkos::View<DataType, Properties...>;
+  using MemSpace = typename ViewType::memory_space;
   static_assert(ViewType::rank == 1,
                 "Kokkos::sort: currently only supports rank-1 Views.");
+  static_assert(SpaceAccessibility<ExecutionSpace, MemSpace>::accessible,
+                "Kokkos::sort: execution space instance is not able to access "
+                "the memory space of the "
+                "View argument!");
 
-  if (view.extent(0) == 0) {
+  if (view.extent(0) <= 1) {
```

Comment:
This is the only place that is changed to `<=1`, the other places now `have `<=0`.